File: TestAuto/TestAuto/ViewController.m
//
//  ViewController.m
//  TestAuto
//
//  Created by user on 14/11/14.
//  Copyright (c) 2014年 user. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    
    UIView *sView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 0, 0)];
    sView.translatesAutoresizingMaskIntoConstraints = NO;
    sView.backgroundColor = [UIColor blackColor];
    
    [self.view addSubview:sView];

    [self.view addConstraint:[NSLayoutConstraint constraintWithItem:sView attribute:NSLayoutAttributeLeft relatedBy:NSLayoutRelationEqual toItem:self.view attribute:NSLayoutAttributeLeft multiplier:1 constant:0]];
    
    [self.view addConstraint:[NSLayoutConstraint constraintWithItem:sView attribute:NSLayoutAttributeRight relatedBy:NSLayoutRelationEqual toItem:self.view attribute:NSLayoutAttributeRight multiplier:1 constant:0]];
    
    [self.view addConstraint:[NSLayoutConstraint constraintWithItem:sView attribute:NSLayoutAttributeHeight relatedBy:NSLayoutRelationEqual toItem:nil attribute:NSLayoutAttributeHeight multiplier:1 constant:20]];
    
    UIView *navView = [[UIView alloc] init];
    navView.backgroundColor = [UIColor orangeColor];
    navView.translatesAutoresizingMaskIntoConstraints = NO;
    [self.view addSubview:navView];
    
    [self.view addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-0-[navView]-0-|" options:0 metrics:nil views:@{@"navView":navView}]];
    [self.view addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|-20-[navView(44)]" options:0 metrics:nil views:@{@"navView":navView}]];

}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: swift/firstUI/firstUI/ViewController.swift
//
//  ViewController.swift
//  firstUI
//
//  Created by newios on 14-8-1.
//  Copyright (c) 2014年 baihuogou. All rights reserved.
//

import UIKit

class ViewController: UIViewController,UITableViewDelegate,UITableViewDataSource {
                            
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        var myTableView:UITableView = UITableView(frame: self.view.bounds, style: UITableViewStyle.Plain)
        myTableView.delegate = self
        myTableView.dataSource = self
        self.view.addSubview(myTableView)
        
//        var myView = UIView(frame: CGRectMake(0, 0, 100, 100))
//        myView.backgroundColor = UIColor.redColor()
//        self.view .addSubview(myView)
        

        
        var swiftView:mySwiftView = mySwiftView(frame:CGRectMake(100, 100, 100, 100))
        swiftView.backgroundColor = UIColor.grayColor()
        self.view.addSubview(swiftView)
        
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func numberOfSectionsInTableView(tableView: UITableView!) -> Int
    {
        return 1;
    }
    func tableView(tableView: UITableView!, numberOfRowsInSection section: Int) -> Int
    {
        return 30
    }
    func tableView(tableView: UITableView!, cellForRowAtIndexPath indexPath: NSIndexPath!) -> UITableViewCell!
    {
        
        let cellId = "Cell"
        var cell :myCell? = tableView.dequeueReusableCellWithIdentifier(cellId) as? myCell
        if cell == nil
        {
            cell = NSBundle.mainBundle().loadNibNamed("myCell", owner: nil, options: nil)[0] as? myCell
            cell!.setButtonClick({
                (button,aIndexPath) in
                
                    NSLog("%@,%d",button,indexPath.row)
                })
        }
        //cell!.mylable.text = "wangliang"
        return cell
    }
    
    func tableView(tableView: UITableView!, heightForRowAtIndexPath indexPath: NSIndexPath!) -> CGFloat
    {
        return 103
    }
    

}

